packages/core/src/external/carequality/ihe-gateway-v2/outbound/xcpd/process/xcpd-response.ts

```diff
@@ -42,31 +42,74 @@ type IheIdentifier = {
   _root: string | undefined;
 };
 
-export function convertIheAddressToCarequalityAddress(address: IheAddress): CarequalityAddress {
+function convertIheIdentifierToPersonalIdentifier(
+  identifier: IheIdentifier
+): PersonalIdentifier | undefined {
+  if (!identifier?._extension && !identifier?._root) {
+    return undefined;
+  }
+  return {
+    value: identifier?._extension,
+    system: identifier?._root,
+  };
+}
+
+function iheIdentifiersToPersonalIdentifiers(
+  otherIds: IheIdentifier[]
+): PersonalIdentifier[] | undefined {
+  if (!otherIds) {
+    return undefined;
+  }
+  const personalIdentifiers = otherIds
+    .map(convertIheIdentifierToPersonalIdentifier)
+    .filter((id): id is PersonalIdentifier => id !== undefined);
+
+  return personalIdentifiers.length > 0 ? personalIdentifiers : undefined;
+}
+
+export function convertIheAddressToAddress(address: IheAddress): Address | undefined {
+  if (!address?.city && !address?.state && !address?.postalCode) {
+    return undefined;
+  }
+  const line = toArray(address?.streetAddressLine).map(String);
   return {
-    line: toArray(address?.streetAddressLine).filter((l): l is string => Boolean(l)),
+    ...(line.length > 0 && { line }),
     city: address?.city ? String(address?.city) : undefined,
     state: address?.state ? String(address?.state) : undefined,
     postalCode: address?.postalCode ? String(address?.postalCode) : undefined,
     country: address?.country ? String(address?.country) : undefined,
   };
 }
 
-function iheAddressesToAddresses(iheAddresses: IheAddress[]): Address[] | undefined {
+export function isValidAddress(address: Address): boolean {
+  const isStringOrStringArray = (value: unknown): value is string | string[] =>
+    typeof value === "string" ||
+    (Array.isArray(value) && value.length > 0 && value.every(item => typeof item === "string"));
+
+  return (
+    (isStringOrStringArray(address?.line) || address?.line === undefined) &&
+    (typeof address?.city === "string" || address?.city === undefined) &&
+    (typeof address?.state === "string" || address?.state === undefined) &&
+    (typeof address?.postalCode === "string" || address?.postalCode === undefined) &&
+    (typeof address?.country === "string" || address?.country === undefined)
```

Comment:
This can result in an address w/ properties all containing empty string, no?

This would be good to be represented/validated w/ a few unit tests.